src/stcal/alignment/util.py

```diff
@@ -3,7 +3,13 @@
 
 import functools
 import logging
-from typing import TYPE_CHECKING, Protocol
+from typing import TYPE_CHECKING, Protocol, runtime_checkable
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Sequence
+
+    import astropy
+    import numpy  # noqa: ICN001 this is the only way Sphinx can find ndarray
```

Comment:
What's the sphinx error that required this? Is it due to the type hints?

Reply:
The error is `/Users/emolter/repositories/stcal/src/stcal/alignment/util.py:docstring of stcal.alignment.util.compute_fiducial:: WARNING: py:class reference target not found: np.ndarray`

It seems to be due to a sphinx bug, but the problem, strangely, only seems to occur when the _output_ type is abbreviated to `np.ndarray`, not when any of the _inputs_ are `np.ndarray`

Reply:
Thanks. I can also see that error in the current docs build. I'm not sure what's going here either. The sphinx version is pinned:
https://github.com/spacetelescope/stcal/blob/eb8528c21e0ffbbdec89f5e1d7323425850d3b69/pyproject.toml#L34
since:
https://github.com/spacetelescope/stcal/commit/3baee3ba572e55ef77c33b00ae45625a775468b1
I tried a newer version and saw the same issue.

Do we perhaps need a sphinx extension or something to allow the docs to use the `ndarray` type hint? Changes to the code to fix the docs seem less than ideal.

Reply:
I agree, although the code change only has to do with type hints, so maybe it's okay?

Reply:
I'm not familiar enough with type hints and the typical docs tools to know what's going on or if there are ways to fix it that don't involve code changes. I'll mark this as resolved since it seems to fix the issue and hasn't raised any other objections.